lib/pages/my_reports_pages/components/reports_list_widget.dart

```diff
@@ -1,98 +1,545 @@
 import 'package:flutter/material.dart';
+import 'package:google_maps_flutter/google_maps_flutter.dart';
+import 'package:intl/intl.dart';
+import 'package:mosquito_alert_app/api/api.dart';
+import 'package:mosquito_alert_app/models/owcampaing.dart';
 import 'package:mosquito_alert_app/models/report.dart';
 import 'package:mosquito_alert_app/utils/MyLocalizations.dart';
 import 'package:mosquito_alert_app/utils/Utils.dart';
+import 'package:mosquito_alert_app/utils/customModalBottomSheet.dart';
 import 'package:mosquito_alert_app/utils/style.dart';
-import 'package:intl/intl.dart';
 
-class ReportsList extends StatelessWidget {
-  final Function onTap;
+
+class ReportsList extends StatefulWidget {
+  ReportsList({Key? key, required this.reports}) : super(key: key);
   final List<Report> reports;
 
-  ReportsList(this.reports, this.onTap);
+  @override
+  _MyReportsListState createState() => _MyReportsListState(reports: reports);
+}
+
+class _MyReportsListState extends State<ReportsList> {
+  List<Report> reports;
+  GoogleMapController? miniMapController;
+
+  _MyReportsListState({required this.reports});
+
+  String formatCreationTime(String? utcTimeString) {
+    utcTimeString ??= '1970-01-01 00:00';
+    var utcTime = DateTime.parse(utcTimeString);
+    var localTime = utcTime.toLocal();
+    var formattedTime = DateFormat('yyyy-MM-dd HH:mm').format(localTime);
+    
+    return formattedTime;
+  }
+
+  String getTitle(BuildContext context, Report report){
+    switch(report.type) {
+      case 'adult':
+        var response_question_6 = report.responses?.firstWhere(
+          (element) => element!.question == 'question_6');
+
+        if (response_question_6 == null){
+          return '';
```

Comment:
return 'Non identified'. Once approved the code of the new 'adult report report flow', this question will stop being answered, and all reports will rely on this 'if' statement.